crates/provider/src/provider/multicall/inner_types.rs

```diff
@@ -160,9 +160,13 @@ impl<D: SolCall> CallTuple for Dynamic<D> {
         let mut ret = vec![];
         for (idx, res) in results.iter().enumerate() {
             if res.success {
```

Comment:
this should just be `if res.success && let Ok() = D::...` instead of checking for empty return data